doc/stdenv/passthru.chapter.md

```diff
@@ -0,0 +1,280 @@
+# Passthru-attributes {#chap-passthru}
+[]{#var-stdenv-passthru} []{#special-variables} <!-- legacy anchors -->
+
+`passthru` is an attribute set which can be filled with arbitrary values.
+For example:
+
+```nix
+{
+  passthru = {
+    foo = "bar";
+    baz = {
+      value1 = 4;
+      value2 = 5;
+    };
+  };
+}
+```
+
+Much of the passthru's power comes from two features:
+- You can access passthru attributes as if they were set inside a derivation (e.g. `hello.baz.value1`).
+- These values are not passed to the builder, so you can change them without triggering a rebuild.
+
+One example of the passthru's power is how it enables us to associate the package for an interpreter Python or Ruby with a set of modules built for that specific interpreter (e.g., `python.pkgs` and `ruby.gems`) and to build runtime environments with a collection of those modules (e.g., (`python.withPackages`)[#python.withpackages-function] and `ruby.withPackages`).
+<!-- python.pkgs, ruby.gems, and ruby.withPackages are mentioned but don't have dedicated sections to link -->
+
+::: {.note}
+Since `passthru` is meant for values that are useful outside the derivation (e.g. in other derivations), it has no strict schema.
+:::
+
+<!--
+I'm extracting this statement from the old text because I feel like it isn't very clearly-phrased and doesn't help connect with something most people know already.
+I still think we should try to make sure the pattern it models is represented in the list of common attributes, though:
+
+convey a specific dependency of your derivation which contains a program with plugins support.
+Later, others who make derivations with plugins can use passed-through dependency to ensure that their plugin would be binary-compatible with built program.
+-->
+
+## Common passthru-attributes {#sec-common-passthru-attributes}
```

Comment:
With the [new convention](https://github.com/NixOS/nixpkgs/pull/245243), I think this entire section should go into `pkgs/README.md` instead, because it's only useful to Nixpkgs contributors. Though one could also argue that the attributes are _exposed_ to users, so they could use it, but that's really not the intention behind these.

And it seems like this would combine well with https://github.com/NixOS/nixpkgs/tree/master/pkgs#package-tests. I never got around to doing this in the original PR, but it would be great to continue going that route!

Reply:
This PR is roughly intended to force this sort of decision. (My primary goal here is that we find clarity on whether and to what extent the passthru should be documented in the manual and/or repo-internal docs and end up with a razor sharp enough to look at a given attr and decide whether/where to document it). 

As long as we leave with that razor in hand, I won't put out if the result is that we close this PR and do something else.

CC: @roberth, since this PR results from discussion with him.